presto-main/src/test/java/com/facebook/presto/block/TestInterleavedBlock.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.block;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.InterleavedBlockBuilder;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Array;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+
+public class TestInterleavedBlock
+        extends AbstractTestBlock
+{
+    private static final int[] SIZE_0 = new int[] {16, 0, 13, 1, 2, 11, 4, 7};
+    private static final int SIZE_1 = 8;
+    private static final int POSITION_COUNT = SIZE_0.length;
+    private static final int COLUMN_COUNT = 2;
+
+    @Test
+    public void test()
+    {
+        Slice[] expectedValues = createExpectedValues();
+        assertValues(expectedValues);
+        assertValues((Slice[]) alternatingNullValues(expectedValues));
+    }
+
+    private void assertValues(Slice[] expectedValues)
+    {
+        InterleavedBlockBuilder blockBuilder = new InterleavedBlockBuilder(ImmutableList.of(VARCHAR, BIGINT), new BlockBuilderStatus(), 10);
+
+        for (Slice expectedValue : expectedValues) {
+            if (expectedValue == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                blockBuilder.writeBytes(expectedValue, 0, expectedValue.length()).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    @Override
+    protected <T> void assertBlockPosition(Block block, int position, T expectedValue)
+    {
+        assertPositionValue(block, position, expectedValue);
+        assertPositionValue(block.getSingleValueBlock(position), 0, expectedValue);
```

Comment:
Instead of all this code to align the regions, I think you should just all unaligned calls to `getRegion`. Conceptually, there's no reason to disallow it, and since you already have `positionCount` in `InterleavedBlock` it looks easy to allow it in the implementation